src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddAnnotationsCachedReaderPass.php

```diff
@@ -27,7 +28,16 @@ public function process(ContainerBuilder $container)
         // "annotations.cached_reader" is wired late so that any passes using
         // "annotation_reader" at build time don't get any cache
         if ($container->hasDefinition('annotations.cached_reader')) {
-            $container->setAlias('annotation_reader', 'annotations.cached_reader');
+            $reader = $container->getDefinition('annotations.cached_reader');
+            $tags = $reader->getTags();
+
+            if (isset($tags['annotations.cached_reader'][0]['provider'])) {
+                if ($container->hasAlias($provider = $tags['annotations.cached_reader'][0]['provider'])) {
+                    $provider = (string) $container->getAlias($provider);
+                }
+                $container->set('annotations.cached_reader', null);
```

Comment:
not in #21533 (methodMap is empty in ContainerBuilder), but in #19668 yes - yet I think we should not merge it as explained implicityl in #21533 :)